src/main.rs

```diff
@@ -41,6 +41,32 @@ static G_SAVE_JSON: Lazy<Mutex<serde_json::Value>> = Lazy::new(|| {
 });
 static mut G_HELLO_WINDOW: Option<Arc<HelloWindow>> = None;
 
+fn version_compat_check(message: String) {
+    let version_tag = fs::read_to_string("/etc/edition-tag").unwrap_or_else(|_| "desktop".to_string());
```

Comment:
```suggestion
    let version_tag = fs::read_to_string("/etc/edition-tag").unwrap_or( "desktop".to_string());
```

probably would be more sense to use unwrap_or here, we dont do any heavy operations for lazy eval